source/patterns/@aws-solutions-constructs/aws-fargate-kinesisstreams/README.md

```diff
@@ -0,0 +1,120 @@
+
+---
+<!--END STABILITY BANNER-->
+
+| **Reference Documentation**:| <span style="font-weight: normal">https://docs.aws.amazon.com/solutions/latest/constructs/</span>|
+|:-------------|:-------------|
+<div style="height:8px"></div>
+
+| **Language**     | **Package**        |
+|:-------------|-----------------|
+|![Python Logo](https://docs.aws.amazon.com/cdk/api/latest/img/python32.png) Python|`aws_solutions_constructs.aws_fargate_kinesis_stream`|
+|![Typescript Logo](https://docs.aws.amazon.com/cdk/api/latest/img/typescript32.png) Typescript|`@aws-solutions-constructs/aws-fargate-kinesisstreams`|
+|![Java Logo](https://docs.aws.amazon.com/cdk/api/latest/img/java32.png) Java|`software.amazon.awsconstructs.services.fargatekinesisstreams`|
+
+## Overview
+This AWS Solutions Construct deploys an AWS Fargate Service that can put records on an Amazon Kinesis Data Stream.
+
+Here is a minimal deployable pattern definition:
+
+Typescript
+``` typescript
+import { Construct } from 'constructs';
+import { Stack, StackProps } from 'aws-cdk-lib';
+import { FargateToKinesisStreamsProps } from '@aws-solutions-constructs/aws-fargate-kinesisstreams';
+import * as fargate from 'aws-cdk-lib/aws-fargate';
+
+new FargateToKinesisStreams(this, 'FargateToKinesisStreams', {
+  defaultVpcIsIsolated: true,
+  ecrRepositoryArn: "arn:aws:ecr:us-east-1:123456789012:repository/your-ecr-repo",
+});
+```
+
+Python
+``` python
+from aws_solutions_constructs.aws_fargate_kinesis_stream import FargateToKinesisStreams
+from aws_cdk import (
+    aws_fargate as _fargate,
+    aws_kinesis as kinesis,
+    Stack
+)
+from constructs import Construct
+
+FargateToKinesisStreams(self, 'FargateToKinesisStreams',
+                          default_vpc_is_isolated=True,
+                          ecr_repository_arn="arn:aws:ecr:us-east-1:123456789012:repository/your-ecr-repo"
+                       )
+```
+
+Java
+``` java
+import software.constructs.Construct;
+
+import software.amazon.awscdk.Stack;
+import software.amazon.awscdk.StackProps;
+import software.amazon.awscdk.services.fargate.*;
+import software.amazon.awscdk.services.fargate.eventsources.*;
+import software.amazon.awscdk.services.fargate.Runtime;
+import software.amazon.awsconstructs.services.fargatekinesisstreams.*;
+
+new FargateToKinesisStreams(this, "FargateToKinesisStreams", new FargateToKinesisStreamsProps.Builder()
+        .defaultVpcIsIsolated(true)
+        .ecrRepositoryArn("arn:aws:ecr:us-east-1:123456789012:repository/your-ecr-repo")
+        .build());
+```
+
+## Pattern Construct Props
+
+| **Name**     | **Type**        | **Description** |
+|:-------------|:----------------|-----------------|
+| defaultVpcIsIsolated | `boolean` | True if the VPC provisioned by this construct should contain only Isolated Subnets, otherwise False for the provisioned VPC to contain Public/Private Subnets. Note this property is ignored if an existing VPC is specified in the `existingVpc` property. |
```

Comment:
This was called `publicApi` in other fargate-based constructs, which only really makes sense to me for those constructs that actually deployed an API. I took a stab at renaming the property when that isn't the case.

Reply:
The front end of the container is unknown to this container - it could still get attached to an ALB-fargate construct. While publicApi may not be the best name, it is the name in the interface of many existing fargate constructs and one of our tenets is "Constructs are Consistent"

Reply:
(subtext - change it back :-)

Reply:
got it  😂